### PR TITLE
Refactor stream config properties related constants

### PR DESCRIPTION
**Pull request by npawar**
As part of the next steps for the issue: https://github.com/linkedin/pinot/issues/2583, we plan to unify all stream related configs into StreamConfigs and remove StreamProviderConfigs. We will also be introducing stream specific config classes such as KafkaStreamConfig. 
This is the first part of that step, which is moving over configs from CommonConstants into StreamConfigProperties, and separating out kafka stream related configs into KafkaStreamConfigProperties.
The follow up of this would be to clean up StreamConfigs and move all kafka specific fields into a KafkaStreamConfig.

### CODE REVIEW

File: pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/KafkaHighLevelStreamProviderConfig.java [hunk 1]
```diff
@@ -198,19 +134,22 @@ public void init(TableConfig tableConfig, InstanceZKMetadata instanceMetadata, S
...
     StreamConfig kafkaMetadata = new StreamConfig(tableConfig.getIndexingConfig().getStreamConfigs());
-    this.kafkaTopicName = kafkaMetadata.getKafkaTopicName();
+    this.kafkaTopicName = kafkaMetadata.getTopicName();
     this.decodeKlass = kafkaMetadata.getDecoderClass();
     this.decoderProps = kafkaMetadata.getDecoderProperties();
     this.kafkaConsumerProps = kafkaMetadata.getKafkaConsumerProperties();
     this.zkString = kafkaMetadata.getZkBrokerUrl();
-    if (tableConfig.getIndexingConfig().getStreamConfigs().containsKey(Helix.DataSource.Realtime.REALTIME_SEGMENT_FLUSH_SIZE)) {
+
+    String flushThresholdRowsProperty = StreamConfigProperties.SEGMENT_FLUSH_THRESHOLD_ROWS;
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/KafkaHighLevelStreamProviderConfig.java]: final

File: pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/KafkaHighLevelStreamProviderConfig.java [hunk 2]
```diff
@@ -198,19 +134,22 @@ public void init(TableConfig tableConfig, InstanceZKMetadata instanceMetadata, S
...
+    if (tableConfig.getIndexingConfig().getStreamConfigs().containsKey(flushThresholdRowsProperty)) {
       realtimeRecordsThreshold =
-          Integer.parseInt(tableConfig.getIndexingConfig().getStreamConfigs().get(Helix.DataSource.Realtime.REALTIME_SEGMENT_FLUSH_SIZE));
+          Integer.parseInt(tableConfig.getIndexingConfig().getStreamConfigs().get(flushThresholdRowsProperty));
     }
 
-    if (tableConfig.getIndexingConfig().getStreamConfigs().containsKey(Helix.DataSource.Realtime.REALTIME_SEGMENT_FLUSH_TIME)) {
+    String flushThresholdTimeProperty = StreamConfigProperties.SEGMENT_FLUSH_THRESHOLD_TIME;
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/KafkaHighLevelStreamProviderConfig.java]: final

File: pinot-core/src/main/java/com/linkedin/pinot/core/realtime/stream/StreamConfigProperties.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  public static final String STREAM_CONSUMER_TYPES = "consumer.type";
+  public static final String STREAM_CONSUMER_FACTORY_CLASS = "consumer.factory.class.name";
+  public static final String STREAM_CONSUMER_OFFSET_CRITERIA = "consumer.prop.auto.offset.reset";
+  public static final String STREAM_FETCH_TIMEOUT_MILLIS = "fetch.timeout.millis";
+  public static final String STREAM_CONNECTION_TIMEOUT_MILLIS = "connection.timeout.millis";
+  public static final String STREAM_DECODER_CLASS = "decoder.class.name";
+  public static final String DECODER_PROPS_PREFIX = "decoder.prop";
+
+  /**
+   * Time threshold that will keep the realtime segment open for before we convert it into an offline segment
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/com/linkedin/pinot/core/realtime/stream/StreamConfigProperties.java]: "before we complete the segment"

**Comment by npawar** [comment]
Addressed comments from review

**Pull request merged**